chain/network/src/network_protocol/borsh_conv.rs

```diff
@@ -0,0 +1,159 @@
+use crate::network_protocol as mem;
+/// Contains borsh <-> network_protocol conversions.
+use crate::network_protocol::borsh as net;
+use anyhow::bail;
+
+impl TryFrom<&net::Handshake> for mem::Handshake {
+    type Error = anyhow::Error;
+    fn try_from(x: &net::Handshake) -> anyhow::Result<Self> {
```

Comment:
This is infallible, right? Consider implementing `From` instead or perhaps using `type Error = std::convert::Infallible` (or both)?

Reply:
done